pycbc/io/live.py

```diff
@@ -558,4 +564,321 @@ def gracedb_tag_with_version(gracedb, event_id):
+
+    # ** means recursive, so for large directories, this is expensive.
+    # It is not too bad if date_directories is set, as we don't waste time
+    # in directories where there cant be any files.
+    glob_string = f'**/*{id_string}*{num_match}*.hdf'
+    if date_directories:
+        # convert the GPS times into dates, and only use the directories
+        # of those dates to search
+        # Add a day on either side to ensure we get files which straddle
+        # the boundary
+        one_day = datetime.timedelta(days=1)
+        date_check = lalgps.gps_to_utc(gps_start_time) - one_day
+        date_end = lalgps.gps_to_utc(gps_end_time) + one_day
+        matching_files = []
+        while date_check < date_end:
+            date_dir = date_check.strftime(date_directory_format)
+            subdir = os.path.join(directory, date_dir)
+            matching_files_gen = pathlib.Path(subdir).glob(glob_string)
+            matching_files += [f.as_posix() for f in matching_files_gen]
+            date_check += one_day
+    else:
+        # Grab all hdf files in the directory
+        matching_files_gen = pathlib.Path(directory).glob(glob_string)
+        matching_files = [f.as_posix() for f in matching_files_gen]
+
+    # Is the file in the time window?
+    matching_files = [f for f in matching_files
+                      if filter_file(f, gps_start_time, gps_end_time)]
+
+    return sorted(matching_files)
+
+
+def find_trigger_files_from_cli(args):
+    """
+    Wrapper around the find_trigger_files function to use when called using
+    options from the add_live_trigger_selection_options function
+    """
+    return find_trigger_files(
+        args.trigger_directory,
+        args.gps_start_time,
+        args.gps_end_time,
+        id_string=args.file_identifier,
+        date_directories=args.date_directories,
+        date_directory_format=args.date_directory_format
+    )
+
+
+def duration_bins_from_cli(args):
+    """Create the duration bins from CLI options.
+    """
+    if args.duration_bin_edges:
+        # direct bin specification
+        return numpy.array(args.duration_bin_edges)
+    # calculate bins from min/max and number
+    min_dur = args.duration_bin_start
+    max_dur = args.duration_bin_end
+    if args.duration_from_bank:
+        # read min/max duration directly from the bank itself
+        with h5py.File(args.duration_from_bank, 'r') as bank_file:
+            temp_durs = bank_file['template_duration'][:]
+        min_dur, max_dur = min(temp_durs), max(temp_durs)
+    if args.duration_bin_spacing == 'log':
+        return numpy.logspace(
+            numpy.log10(min_dur),
+            numpy.log10(max_dur),
+            args.num_duration_bins + 1
+        )
+    if args.duration_bin_spacing == 'linear':
+        return numpy.linspace(
+            min_dur,
+            max_dur,
+            args.num_duration_bins + 1
+        )
+    raise RuntimeError("Invalid duration bin specification")
```

Comment:
pycbc.io.live feels like the wrong place for this function. Does it make sense to start a pycbc.live.single_fits module instead? Or maybe we should move the old stuff to pycbc.io.gracedb instead, since it is not actually that specific to Live anymore?

Reply:
This suggestion backfired as this is now a huge API-breaking PR :) But fine.